lib/system/file.go

```diff
@@ -63,6 +64,72 @@ func CopyFileWithPerms(dst, src string, perm os.FileMode) error {
 	return nil
 }
 
+// Recursively copy
+// dst must always be a directory
+// src may be either a dir or a file
+func CopyAll(src, dst string) (err error, fileCount uint) {
+	fileCount = 0
+	err = copyAll(src, dst, &fileCount)
+	return err, fileCount
+}
+
+func copyAll(src, dst string, fileCount *uint) (err error) {
+	src = filepath.Clean(src)
+	dst = filepath.Clean(dst)
+
+	log.Debugf("CopyAll %s %s", src, dst)
+	si, err := os.Stat(src)
+	if err != nil {
+		return trace.ConvertSystemError(err)
+	}
+	if si.Mode().IsRegular() {
+		*fileCount++
+		return CopyFile(src, filepath.Join(dst, filepath.Base(src)))
+	}
+
+	_, err = os.Stat(dst)
+	if err != nil && !os.IsNotExist(err) {
+		return trace.ConvertSystemError(err)
+	}
+
+	if err != nil && os.IsNotExist(err) {
```

Comment:
I think in this case it would be clearer to combine this with the above:

```go
if err != nil {
  if !os.IsNotExist(err) {
    return ...
  } 
  err = os.MkdirAll(...)
  ...
}
```